### ISSUE TITLE
bug: Exhaustive run during `bentoml build` on AzureDevOps

### ISSUE THREAD

**Issue by aarnphm** [bentoml/BentoML#2827]
### Describe the bug

cc https://bentoml.slack.com/archives/CKRANBHPH/p1658494302553029

TLDR: When running `bentoml build` locally, it works as expected. However, on AzureDevOps python agent, the process seems to hang.

### To reproduce

Current progress:
- I seem to be able to reproduce the hanging locally, it seems like a bad combo of the bentoml CLI and our anti-malware... I am able to run locally by replacing the CLI commands (bentoml build) by a python script doing the same thing (bentoml.bentos.build_bentofile()), it behaves better...
However, things are still stuck on the AzDO build agent... running python -m trace --trace myscript.py it seems stuff is happening, but I am still digesting it. In any case it seems there is no 'hanging' part so far (but I cannot let the script run forever, I already am at 170MB of trace for letting the bentoml.bentos.build_bentofile() for 5 minutes...) 

### Environment

`bentoml`: 1.0.0
`python`: 3.8.11
`platform`: Linux-3.10.0-1160.66.1.el7.x86_64-x86_64-with-glibc2.2.5
`uid:gid`: 0:0
<details><summary><code>pip_packages</code></summary>

<br>

```
aiofiles==0.8.0
aiohttp==3.8.1
aiosignal==1.2.0
anyio==3.6.1
appdirs==1.4.4
asgiref==3.5.2
async-timeout==4.0.2
attrs==21.4.0
bentoml==1.0.0
build==0.8.0
cattrs==22.1.0
certifi==2022.6.15
chardet==5.0.0
charset-normalizer==2.1.0
circus==0.17.1
click==8.1.3
cloudpickle==2.1.0
commonmark==0.9.1
conda==4.3.16
contextlib2==21.6.0
deepmerge==1.0.1
Deprecated==1.2.13
exceptiongroup==1.0.0rc8
filelock==3.7.1
frozenlist==1.3.0
fs==2.4.16
h11==0.13.0
idna==3.3
imageio==2.20.0
Jinja2==3.1.2
MarkupSafe==2.1.1
multidict==6.0.2
numpy==1.23.1
opencv-python==4.6.0.66
opentelemetry-api==1.9.0
opentelemetry-instrumentation==0.28b0
opentelemetry-instrumentation-aiohttp-client==0.28b0
opentelemetry-instrumentation-asgi==0.28b0
opentelemetry-sdk==1.9.0
opentelemetry-semantic-conventions==0.28b0
opentelemetry-util-http==0.28b0
packaging==21.3
pathspec==0.9.0
pep517==0.12.0
Pillow==9.2.0
pip-tools==6.8.0
prometheus-client==0.13.1
psutil==5.9.1
pycosat==0.6.3
Pygments==2.12.0
pynvml==11.4.1
pyparsing==3.0.9
python-dateutil==2.8.2
python-dotenv==0.20.0
python-json-logger==2.0.4
python-multipart==0.0.5
PyYAML==6.0
pyzmq==23.2.0
requests==2.28.1
rich==12.5.1
ruamel.yaml==0.17.21
ruamel.yaml.clib==0.2.6
schema==0.7.5
simple-di==0.1.5
six==1.16.0
sniffio==1.2.0
starlette==0.20.4
tomli==2.0.1
torch==1.12.0
torchvision==0.13.0
tornado==6.2
typing_extensions==4.3.0
urllib3==1.26.11
uvicorn==0.18.2
wrapt==1.14.1
yarl==1.7.2
```
 </details>

**Comment by tibor-mach** [comment]
I am experiencing a very similar issue. I am simply trying to build the bento from the [tutorial](https://docs.bentoml.org/en/latest/tutorial.html).

I tried it locally on Manjaro linux, then in a devcontainer with Ubuntu 20.04.

It hangs (and uses the CPU fully), `--debug` returns nothing, there are no error logs or anything. I tried completely deleting `bentoml` from the local environment and rebuild the container completely (even deleting the entire docker cache with `docker system prune -a`), nothing seems to help.

All I do is the following:
1. In a virtual environment (or the devcontainer) I run `train.py` (as from the tutorial)
2.  then I try to build the bento with `bentoml build`
3. I also tried writing a python script with `bentoml.bentos.build_bentofile()`, this has the same effect (the process keeps hanging, no errors).

Here are the specifications in the devcontainer:
```
`bentoml`: 1.0.4
`python`: 3.8.10
`platform`: Linux-5.19.1-3-MANJARO-x86_64-with-glibc2.29
`uid:gid`: 1000:1000
<details><summary><code>pip_packages</code></summary>

<br>
```
pip packages:

```
aiohttp==3.8.1
aiosignal==1.2.0
anyio==3.6.1
appdirs==1.4.4
asgiref==3.5.2
async-timeout==4.0.2
attrs==22.1.0
bentoctl==0.3.3
bentoml==1.0.4
build==0.8.0
cattrs==22.1.0
Cerberus==1.3.4
certifi==2022.6.15
charset-normalizer==2.1.1
circus==0.17.1
click==8.1.3
cloudpickle==2.1.0
commonmark==0.9.1
contextlib2==21.6.0
deepmerge==1.0.1
Deprecated==1.2.13
docker==5.0.3
exceptiongroup==1.0.0rc8
frozenlist==1.3.1
fs==2.4.16
h11==0.13.0
idna==3.3
Jinja2==3.1.2
joblib==1.1.0
MarkupSafe==2.1.1
multidict==6.0.2
numpy==1.23.2
opentelemetry-api==1.12.0
opentelemetry-instrumentation==0.33b0
opentelemetry-instrumentation-aiohttp-client==0.33b0
opentelemetry-instrumentation-asgi==0.33b0
opentelemetry-sdk==1.12.0
opentelemetry-semantic-conventions==0.33b0
opentelemetry-util-http==0.33b0
packaging==21.3
pandas==1.4.3
pathspec==0.9.0
pep517==0.13.0
pip-tools==6.8.0
prometheus-client==0.13.1
psutil==5.9.1
Pygments==2.13.0
pynvml==11.4.1
pyparsing==3.0.9
python-dateutil==2.8.2
python-dotenv==0.20.0
python-multipart==0.0.5
pytz==2022.2.1
PyYAML==6.0
pyzmq==23.2.1
requests==2.28.1
rich==12.5.1
schema==0.7.5
scikit-learn==1.1.2
scipy==1.9.1
semantic-version==2.10.0
simple-di==0.1.5
simple-term-menu==0.4.4
six==1.16.0
sniffio==1.2.0
starlette==0.20.4
threadpoolctl==3.1.0
tomli==2.0.1
tornado==6.2
typing_extensions==4.3.0
urllib3==1.26.12
uvicorn==0.18.3
watchfiles==0.16.1
websocket-client==1.4.0
wrapt==1.14.1
yarl==1.8.1
```

</details>

```

I also have all the code [here](https://github.com/tibor-mach/test-bento-azure-functions), but it is mostly just the stuff from the tutorial.

**Comment by OlivierBondu** [comment]
Hi guys,

as discussed on slack, I confirm here that this behaviour was due to a virtualenv folder laying around... and that ended up being crawled by bentoML, taking forever (>1h) and causing the entire job to timeout.
Putting the virtualenv outside of the build context (parent directory) solved the issue and the command runs successfully in a couple of seconds as expected.

Probably some improvements in documentation could help?

Thanks for the help!

**Comment by adrianlyjak** [comment]
ah, I had the same problem. Might still be an opportunity here to improve something about this experience that makes the problem more obvious:
Perhaps
- logging something about file processing periodically after a threshold of time is passed
- improving whatever the bottleneck is here about analyzing the imported python from the venv directory. This is a fairly small project, that just has a few dependencies installed:
  ```
  BentoML==1.0.7
  gensim==4.2.0
  pytest==7.1.3
  pytest-cov
  ```

**Comment by aarnphm** [comment]
I will reopen this issue and @jjmachan can you take a look at this when you have time?
